### ISSUE TITLE
Feature request: Cartesian product of point clouds

### ISSUE THREAD

**Issue by nomeata** [PDAL/PDAL#3861]
For one of my projects (related to 3D engraving in glass) I need to create translated copies of points: Create a copy 0.3mm below, or replace every point by a filled sphere of points.

Both operations could be described as an operation acting on two point clouds `S1` and `S2`, replacing each point in S1 with a copy of S2 translated to that position:

\[
S1+S2 = { (x+x', y+y', z+z') | (x,y,z) ∈ S1, (x',y',z') ∈ S2 }
\]

Would that be an operation that would be in scope for a pdal filter?

**Comment by hobu** [comment]
Yes, totally in scope for a filter, but if you needed it RIGHT NOW, you could probably write it in Numpy with [`filters.python`](https://pdal.io/en/stable/stages/filters.python.html)

**Comment by nomeata** [comment]
I am not urgently blocked on this (although I failed to use the Python plugin due to https://github.com/NixOS/nixpkgs/issues/190444), but if I come back to this eventually I might enjoy rewriting it more directly :-)

@chambjj, maybe, but I'd have to use one `translation` filter for each point in S2, or did I misunderstand you?

**Comment by chambbj** [comment]
I think I just conflated your needs. You spoke of copying a point to 0.3 mm below, and then you mentioned replacement in your example - I interpreted that as moving points.
